src/lib/mormot.lib.zstd.pas

```diff
@@ -0,0 +1,394 @@
+/// low-level access to the Zstandard/ZSTD API
+// - this unit is a part of the Open Source Synopse mORMot framework 2,
+// licensed under a MPL/GPL/LGPL three license - see LICENSE.md
+unit mormot.lib.zstd;
+
+
+{
+  *****************************************************************************
+
+   Cross-Platform and Cross-Compiler Zstandard (ZSTD) API
+   - Low-Level Zstandard API Process
+   - TAlgoZSTD High-Level Algorithms
+

```

Comment:
What is missing here is a `THttpSocketCompress` kind of wrapper function, to be used with HTTP client and server classes of the framework.

Reply:
e.g. for `THttpServerGeneric.RegisterCompress()`

Reply:
And this function could actually use `ZSTD_findDecompressedSize()` to uncompress the data, because it works on whole memory blocks.